code/modules/shuttle/computers/dropship_computer.dm

```diff
@@ -0,0 +1,359 @@
+		ui.open()
+
+/obj/structure/machinery/computer/shuttle/dropship/flight/ui_status(mob/user, datum/ui_state/state)
+	. = ..()
+	if(inoperable())
+		return UI_CLOSE
+	if(disabled)
+		return UI_UPDATE
+	if(!skip_time_lock && world.time < SSticker.mode.round_time_lobby + SHUTTLE_TIME_LOCK)
+		to_chat(user, SPAN_WARNING("The shuttle is still undergoing pre-flight fueling and cannot depart yet. Please wait another [round((SSticker.mode.round_time_lobby + SHUTTLE_TIME_LOCK-world.time)/600)] minutes before trying again."))
+		return UI_CLOSE
+
+/obj/structure/machinery/computer/shuttle/dropship/flight/ui_state(mob/user)
+	var/obj/docking_port/mobile/marine_dropship/shuttle = SSshuttle.getShuttle(shuttleId)
+	if(shuttle.is_hijacked)
+		return GLOB.never_state
+	return GLOB.not_incapacitated_and_adjacent_strict_state
+
+/obj/structure/machinery/computer/shuttle/dropship/flight/ui_static_data(mob/user)
+	. = ..(user)
+	var/obj/docking_port/mobile/shuttle = SSshuttle.getShuttle(shuttleId)
+	// we convert the time to seconds for rendering to ui
+	.["max_flight_duration"] = shuttle.callTime / 10
+	.["max_pre_arrival_duration"] = shuttle.prearrivalTime / 10
+	.["max_refuel_duration"] = shuttle.rechargeTime / 10
+	.["max_engine_start_duration"] = shuttle.ignitionTime / 10
+	.["door_data"] = list("port", "starboard", "aft")
+
+/obj/structure/machinery/computer/shuttle/dropship/flight/attack_hand(mob/user)
+	. = ..(user)
+	if(.)
+		return TRUE
+
+	// if the dropship has crashed don't allow more interactions
+	var/obj/docking_port/mobile/marine_dropship/shuttle = SSshuttle.getShuttle(shuttleId)
+	if(shuttle.mode == SHUTTLE_CRASHED)
+		to_chat(user, SPAN_NOTICE("\The [src] is not responsive"))
+		return
+
+	if(!shuttle.is_hijacked)
+		tgui_interact(user)
+
+/obj/structure/machinery/computer/shuttle/dropship/flight/proc/groundside_alien_action(mob/living/carbon/xenomorph/xeno)
+	if(!skip_time_lock && world.time < SSticker.mode.round_time_lobby + (SHUTTLE_TIME_LOCK * 2))
+		playsound(loc, 'sound/machines/terminal_error.ogg', KEYBOARD_SOUND_VOLUME, 1)
+		to_chat(xeno, SPAN_WARNING("No shuttle detected, try again in [round((SSticker.mode.round_time_lobby + (SHUTTLE_TIME_LOCK * 2)-world.time)/600)] minutes."))
+		return
+
+	var/marine_count_planet = SSticker.mode.count_marines(SSmapping.levels_by_trait(ZTRAIT_GROUND))
+	if(!skip_time_lock && (marine_count_planet > 10))
+		playsound(loc, 'sound/machines/terminal_error.ogg', KEYBOARD_SOUND_VOLUME, 1)
+		to_chat(xeno, SPAN_WARNING("There are too many marines near the hive."))
+		return
+
+	var/obj/docking_port/mobile/shuttle = SSshuttle.getShuttle(shuttleId)
+	if(linked_lz)
+		playsound(loc, 'sound/machines/terminal_success.ogg', KEYBOARD_SOUND_VOLUME, 1)
+		if(shuttle.mode == SHUTTLE_IDLE && !is_ground_level(shuttle.z))
+			SSshuttle.moveShuttle(shuttleId, linked_lz, TRUE)
+		else
+			to_chat(xeno, "The screen reads T-[shuttle.timeLeft(10)].")
```

Comment:
This needs:
- An announcement of some sort when the dropship is successfully called down;
- A differentiation between "could not call down because it is not ready yet, wait [timeLeft] seconds" and "it is already in flight, have patience".

Reply:
Sounds good. Will work on this.